jax/numpy/lax_numpy.py

```diff
@@ -242,10 +242,60 @@ def _promote_args_like(op, *args):
 def _constant_like(x, const):
   return onp.array(const, dtype=_dtype(x))
 
+
+def update_numpydoc(docstr, fun, op):
+  if docstr[0] == ' ' or docstr[1] == ' ':
+    lines = docstr.split('\n')
+    for idx, line in enumerate(lines):
+      lines[idx] = line.replace('    ', '', 1)
+    docstr = '\n'.join(lines)
+
+  try:
+    param_list = [arg for arg in fun.__code__.co_varnames]
+    param_list = [arg for arg in param_list if arg in op.__code__.co_varnames]
+  except AttributeError:
+    begin_idx = docstr.find(fun.__name__ + '(') + len(fun.__name__) + 1
+    end_idx = docstr.find(')', begin_idx)
+    param_list = docstr[begin_idx:end_idx].split(', ')
+    for idx, p in enumerate(param_list):
+      eq_idx = p.find('=')
+      if eq_idx == -1:
+        eq_idx = len(p)
+      param = p[:eq_idx]
+      if param[0] == '[':
+        param = param[1:]
+      if param[-1] == ']':
+        param = param[:-1]
+      if param not in op.__code__.co_varnames:
+        param_list[idx] = ''
+    param_list = [param for param in param_list if param != '']
+    parameters = ', '.join(param_list)
+    docstr = docstr[:begin_idx] + parameters + docstr[end_idx:]
+
+  begin_idx = docstr.find("Parameters")
+  begin_idx = docstr.find("--\n", begin_idx) + 2
+  end_idx = docstr.find("Returns", begin_idx)
+
+  parameters = docstr[begin_idx:end_idx]
+  param_list = parameters.replace('\n    ', '@@').split('\n')
+  for idx, p in enumerate(param_list):
+    param = p[:p.find(' : ')].split(", ")[0]
+    if param not in op.__code__.co_varnames:
+      param_list[idx] = ''
+  param_list = [param for param in param_list if param != '']
+  parameters = '\n'.join(param_list).replace('@@', '\n    ')
+  return docstr[:begin_idx + 1] + parameters + docstr[end_idx - 2:]
+
 _numpy_signature_re = re.compile(r'^([\w., ]+=)?\s*[\w\.]+\(.*\)$')
 
-def _wraps(fun):
-  """Like functools.wraps but works with numpy.ufuncs."""
+def _wraps(fun, update_doc=True):
+  """Like functools.wraps but works with numpy.ufuncs.
+    Parameters:
+      fun: The function being wrapped
+      update_doc: Whether you need to update the documentation, 
```

Comment:
Once we resolve the discussion above about what update_numpydoc does, we should find a clearer description of this parameter, e.g., "whether to transform the numpy docstring to remove references of parameters that are supported by the numpy version but not the JAX version. If False, include the numpy docstring verbatim."

Reply:
I used your example description, I think it's clear enough, do you agree?

Reply:
Thanks